rust/src/sans_io/read.rs

```diff
@@ -1101,4 +1102,31 @@ mod tests {
         }
         Ok(())
     }
+
+    #[test]
+    fn test_decompression_does_not_fail() {
+        let mut f = std::fs::File::open("tests/data/break_zstd_decompression.mcap")
```

Comment:
Nit: Could we rename this test and/or mcap file to more clearly indicate the corner case that it's tickling?